espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/Finder.java

```diff
@@ -75,20 +76,17 @@ public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> u
             response.setValue(element);
             response.setSessionId(uriParams.get("sessionId"));
             return response;
-        } catch (NoMatchingViewException e) {
-            response.setResponse(new Appium());
-            return response;
         } catch (InvalidStrategyException e) {
-            response.setResponse(new Appium());
-            return response;
+            return new BadRequestResponse(e.getMessage());
         } catch (ServerErrorException e) {
-            response.setResponse(new Appium());
-            return response;
+            return new InternalErrorResponse(e.getMessage());
+        } catch (ElementNotFoundException e) {
+            return new BadRequestResponse("Could not find element with " + strategy.getStrategyName() + ": " + selector);
```

Comment:
I'll go through these later and fix them. There's a bunch like this one. I just don't want to create any merge conflicts